src/NServiceBus.Core/Pipeline/Incoming/ReceiveFeature.cs

```diff
@@ -20,9 +22,15 @@ protected internal override void Setup(FeatureConfigurationContext context)
             context.Pipeline.Register("TransportReceiveToPhysicalMessageProcessingConnector", b => b.Build<TransportReceiveToPhysicalMessageProcessingConnector>(), "Allows to abort processing the message");
             context.Pipeline.Register("LoadHandlersConnector", b => b.Build<LoadHandlersConnector>(), "Gets all the handlers to invoke from the MessageHandler registry based on the message type.");
 
-            context.Pipeline.Register("ExecuteUnitOfWork", new UnitOfWorkBehavior(), "Executes the UoW");
-            context.Pipeline.Register("MutateIncomingTransportMessage", new MutateIncomingTransportMessageBehavior(), "Executes IMutateIncomingTransportMessages");
-            context.Pipeline.Register("MutateIncomingMessages", new MutateIncomingMessageBehavior(), "Executes IMutateIncomingMessages");
+            var hasUnitsOfWork = context.Container.HasComponent<IManageUnitsOfWork>();
+            context.Pipeline.Register("ExecuteUnitOfWork", new UnitOfWorkBehavior(hasUnitsOfWork), "Executes the UoW");
+
+            var hasIncomingTransportMessageMutators = context.Container.HasComponent<IMutateIncomingTransportMessages>();
```

Comment:
I'd appreciate the insight into the decision here. I tend to try avoid conditionals where possible as it tends to make the debugging experience trickier, so I'd like to understand why doing if-checks and actually invoking extra methods is a good idea.